legend-engine-xt-serviceStore-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/extensions/IServiceStoreGrammarParserExtension.java

```diff
@@ -30,7 +38,21 @@ static List<IServiceStoreGrammarParserExtension> getExtensions()
         return Lists.mutable.withAll(ServiceLoader.load(IServiceStoreGrammarParserExtension.class));
     }
 
-    default List<Function<String, SecurityScheme>> getExtraSecuritySchemesParsers()
+    static SecurityScheme process(SpecificationSourceCode code, List<Function<SpecificationSourceCode, SecurityScheme>> processors)
+    {
+        return ListIterate
+                .collect(processors, processor -> processor.apply(code))
+                .select(Objects::nonNull)
+                .getFirstOptional()
+                .orElseThrow(() -> new EngineException("Unsupported security scheme type '" + code.getType() + "'", code.getSourceInformation(), EngineErrorType.PARSER));
+    }
+
+    default List<Function<SpecificationSourceCode, SecurityScheme>> getExtraSecuritySchemesParsers()
+    {
+        return Collections.emptyList();
+    }
+
+    default List<Function2<String, Map<String,SecurityScheme>, SecuritySchemeRequirement>> getExtraSecurityParsers()
```

Comment:
Lets have the original extension only with Deprecated tag that way we would be backward compatible.
for new one you can use the name `getExtraSecuritySchemeParsers` (Schemes without 's')